relay-metrics/src/aggregation.rs

```diff
@@ -1806,15 +1913,60 @@ impl Aggregator {
                             "Failed to flush the buckets, dropping {batch_size} buckets: {err}."
                         );
                     }
-                });
+                }
+                num_batches += 1;
+                relay_statsd::metric!(
+                    histogram(MetricHistograms::BatchesPerPartition) = num_batches as f64,
+                );
             }
         }
-
         relay_statsd::metric!(histogram(MetricHistograms::BucketsFlushed) = total_bucket_count);
     }
+
+    async fn handle_message(&mut self, msg: Aggregator) {
+        match msg {
+            // This is the `HealthCheck` request where the `AsyncResponse` is expected.
+            Aggregator::AcceptsMetrics(AcceptsMetrics, sender) => {
+                let result = !self
+                    .cost_tracker
+                    .totals_cost_exceeded(self.config.max_total_bucket_bytes);
+                sender.send(result);
```

Comment:
Please introduce individual `handle_xyz` (e.g. `handle_merge_buckets`) methods and dispatch from `handle_message`. 

This is the pattern used in other services, too. We use this because it helps with navigating and reading code, especially when there are many messages.

Here is an example with both kinds of messages: https://github.com/getsentry/relay/blob/95f154409f236ecc8089a4b19534997625a21735/relay-server/src/actors/test_store.rs#L128-L133